### ISSUE TITLE
Hubs client does not load in WebKit Webviews (Chrome/Slack on iOS)

### ISSUE THREAD

**Issue by brianpeiris** [mozilla/hubs#181]
Could also be a problem with Android Slack? We'll have to detect and bust out of the webview into Safari.

**Comment by robertlong** [comment]
How do you get the client to open up in this webview? Links always open up in Firefox for me.

**Comment by robertlong** [comment]
Hmm, I still can't reproduce this on Android or iOS. They both open in a browser for me.

**Comment by robertlong** [comment]
Ok, there's a setting in Slack to set what browser to use when opening links. Mine was set to Safari on iOS and Firefox on Android.

I don't think there is any way to bust out of the webview on iOS.

On Android this works:
```
<html>
  <head></head>
  <script>
    document.location = "intent://google.com#Intent;scheme=http;action=android.intent.action.VIEW;end;";
  </script>
  <body>
  </body>
</html>
```

Should we do this hack on Android or just try to show a message to open the page in Safari/Chrome/Firefox if we detect that they are in a webview?

**Comment by robertlong** [comment]
Removing this from the MVP as there isn't a clean way to do this.

**Comment by cvan** [comment]
I encounter this same issue using Chrome for iOS too. Investigating a solution here.

**Comment by brianpeiris** [comment]
I think we settled on showing a message at least, unless there is a proper solution.

**Comment by cvan** [comment]
> I think we settled on showing a message at least, unless there is a proper solution.

haven't note Remote Debugged it yet, what have you determined the errors/issues are? any Console/Sentry errors?

**Comment by cvan** [comment]
wow, so I just spent hours trying to figure out how to remotely debug on iOS with anything other than Safari. that includes any WebKit webviews (i.e., those used in Chrome, Slack, etc.).

I've tried the following with zero success:

- [`google/ios-webkit-debug-proxy`](https://github.com/google/ios-webkit-debug-proxy)
- [building an iOS WebView app in XCode](https://stackoverflow.com/a/25817718) (simulator got stuck here, and it wasn't exposed from Safari's `Develop > iPhone Simulator` menu)
    
![image](https://user-images.githubusercontent.com/203725/39460367-c2324d2a-4cb7-11e8-8467-a7b95e12c162.png)
- [`RemoteDebug/remotedebug-ios-webkit-adapter`](https://github.com/RemoteDebug/remotedebug-ios-webkit-adapter)
- [`artygus/webkit-webinspector`](https://github.com/artygus/webkit-webinspector)
- [Firefox Dev Tools' `debugger.html`](http://firefox-dev.tools/debugger.html/)
- [`weinre`](https://stackoverflow.com/questions/11262236/ios-remote-debugging/22047495#22047495) (no out-of-the-box HTTPS support!)
- [`vorlon.js`](http://vorlonjs.com/) (again, sigh)

and it's unclear from [RemoteDebug.org](https://remotedebug.org/) what's going on either.

long story short: remotely debugging web pages in Safari is easy. debugging pages in Chrome for iOS/any other app with a webview: IDK! if you know, please send help! 👍

**Comment by cvan** [comment]
@robertlong I've been hearing some people encounter the issue in vanilla iOS. can you provide any info on what you've discovered so far?

**Comment by gfodor** [comment]
we won't be supporting this due to the issues mentioned above, but @InfiniteLee pushed a UX update that will guide the user out of the webview.
